libs/remote_storage/src/azure_blob.rs

```diff
@@ -219,21 +220,26 @@ impl RemoteStorage for AzureBlobStorage {
     }
     async fn upload(
         &self,
-        mut from: impl AsyncRead + Unpin + Send + Sync + 'static,
+        from: impl futures::stream::Stream<Item = std::io::Result<bytes::Bytes>> + Send + Sync + 'static,
         data_size_bytes: usize,
         to: &RemotePath,
         metadata: Option<StorageMetadata>,
     ) -> anyhow::Result<()> {
         let _permit = self.permit(RequestKind::Put).await;
         let blob_client = self.client.blob_client(self.relative_path_to_name(to));
 
-        // TODO FIX THIS UGLY HACK and don't buffer the entire object
-        // into RAM here, but use the streaming interface. For that,
-        // we'd have to change the interface though...
-        // https://github.com/neondatabase/neon/issues/5563
-        let mut buf = Vec::with_capacity(data_size_bytes);
-        tokio::io::copy(&mut from, &mut buf).await?;
-        let body = azure_core::Body::Bytes(buf.into());
+        let from: std::pin::Pin<
+            Box<
+                dyn futures::stream::Stream<Item = std::io::Result<bytes::Bytes>>
```

Comment:
this is formatted in a very ugly way. Could you try to shorten it by a) importing `Stream` and also maybe `Bytes`?

Reply:
+1, I'd also do the same with `Pin`

Reply:
> this is formatted in a very ugly way

but it is ugly :) Can pretty it up.

Reply:
2f56cfcfa after next push.

Reply:
Argh, I of course rebased again. d293cafb2